### PR TITLE
Reduce SN load in includes/socialnet.php

### PR DESCRIPTION
**Pull request by senky**
http://phpbbsocialnetwork.com/viewtopic.php?f=44&t=4743

### CODE REVIEW

**Comment by Culprit-CZ** [comment]
MA

**Comment by Kamahl19** [comment]
MA

File: root/socialnet/includes/socialnet.php [hunk 1]
```diff
@@ -271,9 +271,23 @@ function purge_friends($user_id = 0)
 	{
 		global $user, $cache;
```
Comment (woipi90) [inline root/socialnet/includes/socialnet.php:272]: sry to see only now define $phpbb_root_path here..

File: root/socialnet/includes/socialnet.php [hunk 2]
```diff
@@ -271,9 +271,23 @@ function purge_friends($user_id = 0)
...
 
-		$user_id = ($user_id == 0) ? $user->data['user_id'] : $user_id;
-		$cache->destroy($this->friendsCacheName . $user_id);
-		$cache->destroy($this->friendsCacheNameMutual . $user_id);
+		if ($user_id == 'all')
+		{
+			$files_to_be_purged = glob($phpbb_root_path . 'cache/*' . $socialnet->friendsCacheNameMutual . '*');
```
Comment (woipi90) [inline root/socialnet/includes/socialnet.php:276]: maybe u wanna use $this here..

**Pull request merged**